### PR TITLE
new averaging option for fix temp/rescale

### PR DESCRIPTION
**Pull request by sjplimp**
## Purpose

Add a new "ave" option to fix temp/rescale to allow the current thermal temperature used for velocity rescaling to be the average thermal temperature across all grid cells.  Rather than the thermal terperature for each individual cell, as is the current case. 

## Author(s)

Steve

## Backward Compatibility

N/A

## Implementation Notes

_Provide any relevant details about how the changes are implemented, how correctness was verified, how other features - if any - in SPARTA are affected_

## Post Submission Checklist

_Please check the fields below as they are completed_
- [ ] The feature or features in this pull request is complete
- [ ] Suitable new documentation files and/or updates to the existing docs are included
- [ ] One or more example input decks are included
- [ ] The source code follows the SPARTA formatting guidelines

## Further Information, Files, and Links

_Put any additional information here, attach relevant text or image files, and URLs to external sites (e.g. DOIs or webpages)_

### CODE REVIEW

**Comment by sjplimp** [comment]
@stanmoore1 Please wait for Michael to test this.  Once it is ready, this option can be added to the Kokkos version as well.

File: src/fix_temp_rescale.cpp [hunk 1]
```diff
@@ -149,3 +189,143 @@ void FixTempRescale::end_of_step()
...
+  double invtotmass,t_one;
+  double *v;
+
+  bigint n_current_mine = 0.0;
+  double t_current_mine = 0.0;
+  
+  for (int icell = 0; icell < nglocal; icell++) {
+    if (cells[icell].nsplit > 1) continue;
+
+    count = cinfo[icell].count;
```
Comment (stanmoore1) [inline src/fix_temp_rescale.cpp:229]: Count should be `int` instead of `double`?

File: src/fix_temp_rescale.cpp [hunk 2]
```diff
@@ -149,3 +189,143 @@ void FixTempRescale::end_of_step()
...
+    // t_one = thermal T of particles in the cell
+    // vcom = COM velocity of particle in the cell
+    // if cell has <= 1 particle: set t_one = t_target, vcom = zero
+    // likewise if t_one = 0.0: set t_one = t_target, vcom = zero
+    //   corner case when all particles have same velocity
+
+    if (count > 1.0) {
+      invtotmass = 1.0/totmass;
+      t_one = mvsq - (mvx*mvx + mvy*mvy + mvz*mvz)*invtotmass;
+      count = cinfo[icell].count;
```
Comment (stanmoore1) [inline src/fix_temp_rescale.cpp:261]: Is this (`count = cinfo[icell].count`)needed?

**Review by stanmoore1 — CHANGES_REQUESTED** [review]

**Comment by sjplimp** [comment]
just pushed a modified version which I think addresses both Qs

**Review by stanmoore1 — APPROVED** [review]

**Pull request merged**